### ISSUE TITLE
List collaborators does not expose the affiliation parameter (or preview API)

### ISSUE THREAD

**Issue by shiftkey** [octokit/octokit.net#2036]
https://developer.github.com/v3/repos/collaborators/#list-collaborators

https://github.com/octokit/octokit.net/blob/8b263cdc0f80102d358bd78dcb58151537e7e847/Octokit/Clients/RepoCollaboratorsClient.cs#L23-L86

Two things need to be added here:

 - support for `affiliation` - this should be a new enum and not our current [`RepositoryAffiliation`](https://github.com/octokit/octokit.net/blob/74dc51a6f567395d0c46d97f7270f959d671573e/Octokit/Models/Request/RepositoryRequest.cs#L170).

<img width="674" src="https://user-images.githubusercontent.com/359239/67484729-13f97980-f63f-11e9-8bc3-ec464b758447.png">

 - support for the nested teams preview API,  which has been available for a while

<img width="667" alt="Screen Shot 2019-10-24 at 9 18 45 AM" src="https://user-images.githubusercontent.com/359239/67484874-515e0700-f63f-11e9-96e2-87f264339c80.png">

cc @terrajobst as you were interested in this

**Comment by terrajobst** [comment]
Yes, I'm very much interested in that. I wrote a quick & dirty tool to audit the GitHub permissions in our org. As you can imagine, we have several hundred repos spread over various orgs. Clicking through the UI isn't going to cut it. Also, we want to apply some filters (like age of the repo and whether it's public/private) to limit the noise. Here is the kind of report I came up with:

![image](https://user-images.githubusercontent.com/5169960/67517095-a78a7480-f656-11e9-8353-bb5c3ee61359.png)

Basically, I want to dump all teams and users that have given explicit access on the repo (rather than getting access through a team).

**Comment by shiftkey** [comment]
I think adding overloads for this new enum parameter here is the quickest way to get what you need supported (nested teams aren't dependent on that):

```c#
var results = await client.Repository.Collaborator.GetAll("owner", "name", CollaboratorAffiliation.Direct);
var resultsWithPagination = await client.Repository.Collaborator.GetAll("owner", "name", CollaboratorAffiliation.Direct, apiOptions);
```

**Comment by terrajobst** [comment]
This might also significantly improve performance. Because when transitive members are included, the API basically returns several thousand users per repo.

**Comment by hnrkndrssn** [comment]
> I think adding overloads for this new enum parameter here is the quickest way to get what you need supported
Existing conventions use a `Request` class that extends the `RequestParameters` class (to give an easy way to convert it to a parameters dictionary). Should we deviate from this convention here? _I've assumed that we should follow the existing convention as I've been spelunking on adding this._
